source/common/http/conn_manager_impl.cc

```diff
@@ -1695,6 +1695,15 @@ void ConnectionManagerImpl::ActiveStream::encodeHeadersInternal(ResponseHeaderMa
     ENVOY_STREAM_LOG(debug, "drain closing connection", *this);
   }
 
+  // The BadRequest error code indicates there has been a messaging error.
```

Comment:
Just so I understand: does this code path not get hit if the upstream returns `BadRequest`?

Reply:
It does.  I can change it so only Envoy invalid messaging triggers closes if we prefer that.

Reply:
I've definitely seen people use BadRequest to indicate a malformed message body (ie invalid JSON key or whatever) out in the wild, so it might make sense to only close the connection when we know that it actually was related to invalid HTTP messages? 

I'm not sure what the right thing to do here tbh, not clear what the scope of "HTTP messaging" is